docs/content/reference/exploiters/snmp.md

```diff
@@ -0,0 +1,107 @@
+---
+title: "SNMP"
+draft: false
+description: "Reference material for the SNMP exploiter"
+pre: "<i class='fas fa-network-wired'></i> "
+---
+{{< table_of_contents >}}
+
+## Configuration
+
+<!--
+This documentation was autogenerated by passing the plugin's config-schema.json
+through https://github.com/adobe/jsonschema2md. It was then modified by hand to
+remove extraneous information.
+-->
+
+| Property                                                           | Type      | Required | Nullable       |
+| :----------------------------------------------------------------- | :-------- | :------- | :------------- |
+| [agent\_binary\_download\_timeout](#agent_binary_download_timeout) | `number`  | Optional | cannot be null |
+| [snmp\_request\_timeout](#snmp_request_timeout)                    | `number`  | Optional | cannot be null |
+| [snmp\_retries](#snmp_retries)                                     | `integer` | Optional | cannot be null |
+
+### agent\_binary\_download\_timeout
+
+The maximum time (in seconds) to wait for a successfully exploited SNMP client to download the agent binary.
+
+`agent_binary_download_timeout`
+
+* is optional
+
+* Type: `number`
+
+* cannot be null
+
+#### agent\_binary\_download\_timeout Constraints
+
+**minimum (exclusive)**: the value of this number must be greater than: `0`
+
+#### agent\_binary\_download\_timeout Default Value
+
+The default value is:
+
+```json
+60
+```
+
+### snmp\_request\_timeout
+
+The maximum time (in seconds) to wait for a response from an SNMP client.
+
+`snmp_request_timeout`
+
+* is optional
+
+* Type: `number`
+
+* cannot be null
+
+#### snmp\_request\_timeout Constraints
+
+**minimum (exclusive)**: the value of this number must be greater than: `0`
+
+#### snmp\_request\_timeout Default Value
+
+The default value is:
+
+```json
+0.5
+```
+
+### snmp\_retries
+
+The number of times to retry an SNMP request before giving up.
+
+`snmp_retries`
+
+* is optional
+
+* Type: `integer`
+
+* cannot be null
+
+#### snmp\_retries Constraints
+
+**minimum**: the value of this number must greater than or equal to: `0`
+
+#### snmp\_retries Default Value
+
+The default value is:
+
+```json
+3
+```
+
+## ???
```

Comment:
I'm having trouble thinking of an appropriate name for this section.

Reply:
Yeah, me too. After reading through it, I'm thinking it's explanation and not reference. LMK if you agree with my reasoning.

![image](https://github.com/guardicore/monkey/assets/19957806/e1f56137-9bbb-40d4-8a17-181e7a3dd25f)
![image](https://github.com/guardicore/monkey/assets/19957806/b6087321-23bb-4a85-943a-16d52877bf8b)

I think the information is theoretical. In other words, it doesn't provide any concrete steps. In addition, I think the information is for study, not for work. Information about SNMP protocols provides me with context. Information about what machines are vulnerable and how SNMP protocols differ might be the kind of thing that could be "read in the bath". To put it yet another way, even though it does provide some information about _how_ the exploiter works, it's not likely to be information I would consult while configuring the exploiter (work). Instead, I might consult it when I was trying to understand (study) the capabilities of Infection Monkey.

I think a table for supported SNMP versions is warranted and we can use this text when we write the explanation of the features.

![image](https://github.com/guardicore/monkey/assets/19957806/2cb53c95-2ade-434e-9de4-e0d99a512ed0)

Thoughts?

Reply:
> I think a table for supported SNMP versions is warranted and we can use this text when we write the explanation of the features.

That sounds good to me.